### DISCUSSION TITLE
Server API endpoints (even with nothing in it) are too slow (150ms response time)

### DISCUSSION THREAD

**Discussion by madebyfabian** [nuxt/nuxt#16265] [Questions]
Hey there, I am currently looking to optimize the speed of my api routes, and in general, they are pretty slow, so I thought of removing everything. So basically, with everything removed, an empty api request still takes about 150ms, which I find extremely long. Here is the example code:

```ts
export default defineEventHandler(async event => {
    return true
})
```

I push to vercel with no further config.

Does anyone else have the issue? Or is 150ms just acceptable for an api request? I rember setting up basic nodejs servers in the past that would do 10-20ms for (almost) empty requests.

![Bildschirmfoto_2022-11-27_um_12 44 55](https://user-images.githubusercontent.com/32813692/204245561-d409600a-9866-4265-8842-113a61d48618.jpg)

**Comment by TheAlexLichter** [comment]
<img width="1786" alt="Dev Tools for a simple API endpoint showing 3ms response time" src="https://user-images.githubusercontent.com/640208/204249166-8c2f98ac-3b37-4801-9f47-54e69d0af907.png">


Please ensure you benchmark in production, not during dev mode 😋

Having an easy 3ms for a local node server via `pnpm build && node .output/server/index.mjs`

**Reply by madebyfabian** [reply]
@manniL Thanks for your answer. These are indeed benchmarks from production. You can try them out at this url
https://twitter-clone-git-temporary-test-routes-madebyfabian.vercel.app/admin/test

Sometimes it's even above 200ms.

**Reply by TheAlexLichter** [reply]
Can you try benchmarking e.g. with a local node server?

You might see cold starts or similar on a platform like vercel.

**Reply by madebyfabian** [reply]
@manniL Locally, for an empty api request like I have in the demo, it's about 7ms. 

(see the `/api/v2/admin/test` route, the other one has some DB queries in it)

## Local
![Bildschirm­foto 2022-11-28 um 13 55 16](https://user-images.githubusercontent.com/32813692/204283238-301a2ad5-fd4f-42ea-97b4-be50771289ae.jpg)

## Vercel
![Bildschirm­foto 2022-11-28 um 13 55 48](https://user-images.githubusercontent.com/32813692/204283259-0fc92245-d81a-43db-b79b-972a1dac9da4.jpg)

So it's performing a lot worse in production. So yeah cold boots could be a thing but does this mean that every api request boots a function and then shuts it down? Since I thought maybe the first request would be slow due to cold boot, but others should be significantly faster, which isn't the case.

**Reply by TheAlexLichter** [reply]
Sadly I'm not an expert when it comes to vercel itself (rarely using it for projects). Maybe @danielroe has more input here or an idea why calls could be that slow? 🤔

**Reply by madebyfabian** [reply]
@manniL No worries, thank you for your time! I would love to continue using nuxt3 server api routes with vercel, but let's see. Maybe there is something that can be improved from my side.
